docs/en/stack/data-frames/checkpoints.asciidoc

```diff
@@ -0,0 +1,60 @@
+[role="xpack"]
+[[ml-transform-checkpoints]]
+=== {dataframe-transform-cap} checkpoints
+++++
+<titleabbrev>Checkpoints</titleabbrev>
+++++
+
+beta[]
+
+Each time a {dataframe-transform} examines the source indices and creates or
+updates the destination index, it generates a _checkpoint_.
+
+If your {dataframe-transform} runs only once, there is logically only one
+checkpoint. If your {dataframe-transform} runs continuously, however, it creates
+checkpoints as it ingests and transforms new source data.
+
+For example, a {cdataframe-transform}:
+
+. Checks for changes to source indices.
++
+Using a simple periodic timer, the {dataframe-transform} checks for changes to
+the source indices based on sequence IDs. This check is done based on the
+interval defined in the transform's `frequency` property.
++
+If the source indices remain unchanged or if a checkpoint is already in progress
+then it waits for the next timer.
+
+. Identifies which entities have changed.
++
+The {dataframe-transform} searches to see which entities have changed since the
+last time it checked. The transform's `sync` configuration object identifies a
+time field in the source indices. The transform uses the values in that field to
+synchronize the source and destination indices.
+ 
+. Updates the destination index (the {dataframe}) with the changed entities.
++
+--
+The {dataframe-transform} applies changes related to either new or changed
+entities to the destination index.
+//TBD: This seems to imply it doesn't handle deletions. Do we want to clarify that?
```

Comment:
Handling of deletions should already be in limitations. It would be too much info to add here (and also likely to change in next iteration).